nemo_skills/evaluation/metrics/base.py

```diff
@@ -29,6 +30,92 @@ def get_metrics(self):
     def reset(self):
         pass
 
+    @abc.abstractmethod
+    def get_prediction_results(self, prediction):
+        pass
+
+    def get_majority_at_k(
+        self, agg_mode_dict, predicted_answers, pred_keys=None, predictions=None, prediction_results=None
+    ):
+        """
+        Get majority@k metrics for a given set of prediction results.
+
+        Args:
+            agg_mode_dict (dict): Dictionary to store aggregated metrics.
+            predicted_answers (list): List of generated predictions.
+            pred_keys (Optional[list]): List of keys to aggregate over. If not provided, it will use all keys in the prediction results.
+            predictions (Optional[list]): List of generated predictions. These will go through `get_prediction_results` to get the prediction results.
+            prediction_results (Optional[list]): List of prediction results. If provided, it will ignore the `predictions` argument.
+        """
+        if prediction_results is None:
+            assert predictions is not None, "Either predictions or prediction_results must be provided"
+            prediction_results = [self.get_prediction_results(pred) for pred in predictions]
+
+        if pred_keys is None:
+            pred_keys = prediction_results[0].keys()
+
+        for k in range(len(prediction_results), 1, -1):
```

Comment:
after we change max aggregations logic, can probably have a normal loop here, not reversed to make it simpler